test/gtest/uct/ib/test_dc.cc

```diff
@@ -452,6 +452,28 @@ UCT_DC_INSTANTIATE_TEST_CASE(test_dc)
 class test_dc_flow_control : public test_rc_flow_control {
 public:
 
+    class allow_misordering {
+    public:
+        allow_misordering(uct_test::entity *e)
+        {
+#if UCS_ENABLE_ASSERT
+            m_iface = test_dc::dc_iface(e);
+
+            m_iface->tx.allow_misordering = 1;
+#endif
+        }
+
+        ~allow_misordering()
+        {
+#if UCS_ENABLE_ASSERT
+            m_iface->tx.allow_misordering = 0;
+#endif
+        }
```

Comment:
IMO overcomplication - just set the allow flag once during the test, no need to remove this flag

Reply:
> IMO overcomplication - just set the allow flag once during the test, no need to remove this flag

ok, used single routine